book/image-controller.md

```diff
@@ -1,2 +1,125 @@
 # Image Controller
 
+## Overview
+
+The Image Controller for AppStudio helps set up container image repositories
+for AppStudio Components. It watches Components to take actions according to
+the deletion event and the value set to specific annotations.
+
+External services are able to interact with image controller via these two
+annotations:
+
+- `image.redhat.com/generate`: if set, image controller starts to set up image
+  repository on Quay.io for a Component, or attempts to switch repository
+  visibility if the repository exists already.
+
+- `image.redhat.com/image`: information of created resources is set to this
+  annotation as a JSON string. Error message will also be set to here if there
+  is.
+
+## Set up image repository
+
+To request the controller to setup an image repository, annotate the Component
+CR with `image.redhat.com/generate: '{"visibility": "public"}'` or
+`image.redhat.com/generate: '{"visibility": "private"}'` depending on desired
+repository visibility.
+
+Here is an example that a component is annotated for creating a public image
+repository:
+
+```yaml
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: Component
+metadata:
+  annotations:
+    image.redhat.com/generate: '{"visibility": "public"}'
+  name: billing
+  namespace: image-controller-system
+spec:
+  application: city-transit
+  componentName: billing
+```
+
+As a result, when the setup succeeds,
+
+- A public image repository is created on Quay.io.
+- A robot account is created and associated with the repository properly.
+- A Kubernetes Secret is created alongside the Component CR, to which the
+  robot account token is written out.
```

Comment:
Is this secret also bound to the ServiceAccount?

Reply:
Yes. It is what build-service does.

Reply:
You indicate this at the end of the section. Instead, you can you pull that out and explain it in the overview? If you give a brief explanation of the image controller (i.e. what it is responsible for) and then indicate how it interfaces with other services, then that would be helpful. You would be able to explain conceptually the whole process without the json/yaml blocks in between to separate the contexts.

Reply:
Thanks for your comment. PTAL.